gapic/src/main/com/google/gapid/perfetto/views/ThreadPanel.java

```diff
@@ -412,11 +409,11 @@ public void computeSelection(CombiningBuilder builder, Area area, TimeSpan ts) {
     }
 
     if (startDepth == 0) {
-      builder.add(Selection.Kind.ThreadState,
-          transform(track.getStates(ts), ThreadTrack.StateSlicesBuilder::new));
-      builder.add(Selection.Kind.Cpu, transform(track.getCpuSlices(ts), r -> {
-        r.forEach(s -> state.addSelectedThread(state.getThreadInfo(s.utid)));
-        return new CpuTrack.SlicesBuilder(r);
+      // Extra transform here solving a java generic type recognition problem.
```

Comment:
I've sent you a PR (https://github.com/stellama0208/agi/pull/1) to get rid of this.

Reply:
Thanks a lot for helping solve this complicated generic thing! A lot to learn in this PR!
Somehow on my machine, there's still a little compilation problem after this patch, so in "Address comments #1" I added several more type conversions, hope they are tolerable!